input/system/selfhosted/otel_handler.go

```diff
@@ -105,73 +108,115 @@ func skipDueToK8sFilter(kubernetes *common.KeyValueList, server *state.Server, p
+				var kubernetes *common.KeyValueList
+				hasErrorSeverity := false
+				if l.Body.GetKvlistValue() != nil {
+					for _, v := range l.Body.GetKvlistValue().Values {
+						if v.Key == "logger" {
+							logger = v.Value.GetStringValue()
+						}
+						if v.Key == "record" {
+							record = v.Value.GetKvlistValue()
+						}
+						if v.Key == "kubernetes" {
+							kubernetes = v.Value.GetKvlistValue()
+						}
+						if v.Key == "error_severity" {
+							hasErrorSeverity = true
 						}
 					}
+					// TODO: Support other logger names (this is only tested with CNPG)
+					if logger == "postgres" {
+						// jsonlog wrapped in K8s context (via fluentbit)
+						logLine, detailLine := logLineFromJsonlog(record, logParser)
+						if skipDueToK8sFilter(kubernetes, server, prefixedLogger) {
+							continue
+						}
+
+						parsedLogStream <- state.ParsedLogStreamItem{Identifier: server.Config.Identifier, LogLine: logLine}
+						if detailLine != nil {
+							parsedLogStream <- state.ParsedLogStreamItem{Identifier: server.Config.Identifier, LogLine: *detailLine}
+						}
+					} else if logger == "" && hasErrorSeverity {
+						// simple jsonlog (Postgres jsonlog has error_severity key)
+						logLine, detailLine := logLineFromJsonlog(l.Body.GetKvlistValue(), logParser)
+						parsedLogStream <- state.ParsedLogStreamItem{Identifier: server.Config.Identifier, LogLine: logLine}
+						if detailLine != nil {
+							parsedLogStream <- state.ParsedLogStreamItem{Identifier: server.Config.Identifier, LogLine: *detailLine}
+						}
+					}
+				} else if l.Body.GetStringValue() != "" {
+					// plain log message
+					item := SelfHostedLogStreamItem{}
+					item.Line = l.Body.GetStringValue()
+					item.OccurredAt = time.Unix(0, int64(l.TimeUnixNano))
+					rawLogStream <- item
 				}
 			}
-		})
-		http.ListenAndServe(otelLogServer, nil)
-	}()
+		}
+	}
+	return w
+}
+
+func setupOtelHandler(ctx context.Context, server *state.Server, rawLogStream chan<- SelfHostedLogStreamItem, parsedLogStream chan state.ParsedLogStreamItem, prefixedLogger *util.Logger, opts state.CollectionOpts) error {
+	otelLogServer := server.Config.LogOtelServer
+
+	if otelLogServer != "" {
+		if !util.SliceContains(otelServers, otelLogServer) {
+			serverMux := http.NewServeMux()
+			serverMux.HandleFunc("/v1/logs", func(w http.ResponseWriter, r *http.Request) {
+				otelV1LogHandler(w, r, server, rawLogStream, parsedLogStream, prefixedLogger, opts)
+			})
+
+			go func() {
+				err := http.ListenAndServe(otelLogServer, serverMux)
+				prefixedLogger.PrintInfo("Registered OpenTelemetry log handler on %s", otelLogServer)
+				if err != nil {
+					prefixedLogger.PrintError("Error starting server on %s: %v\n", otelLogServer, err)
+				}
+			}()
+			otelServers = append(otelServers, otelLogServer)
+		} else {
+			prefixedLogger.PrintInfo("OpenTelemetry log handler on %s already registered, skipping. Check your configuration for duplicate entries.", otelLogServer)
```

Comment:
It would be nice to mention the config name (`db_log_otel_server`) here. Uhmm I wasn't able to find an example, but maybe something like "for duplicate entries in db_log_otel_server"...?